kotlin-guice/src/main/kotlin/dev/misfitlabs/kotlinguice4/KotlinModule.kt

```diff
@@ -57,18 +59,33 @@ import dev.misfitlabs.kotlinguice4.internal.KotlinBindingBuilder
  * @since 1.0
  */
 abstract class KotlinModule : AbstractModule() {
-    /** Gets direct access to the underlying [KotlinBinder]. */
-    protected val kotlinBinder: KotlinBinder by lazy {
-        KotlinBinder(binder().skipSources(
-            KotlinAnnotatedBindingBuilder::class.java,
-            KotlinAnnotatedElementBuilder::class.java,
-            KotlinBinder::class.java,
-            KotlinBindingBuilder::class.java,
-            KotlinLinkedBindingBuilder::class.java,
-            KotlinScopedBindingBuilder::class.java
-        ))
+    private class KotlinLazyBinder(private val binderSource: Provider<Binder>) {
```

Comment:
Instead of using `Provider` here what do you think about naming it `delegateBinder` and having the type be `() -> Binder`?

Reply:
that's much better! I still need to get ride of my Java's habits :)